observers/slack_resource.hpp

```diff
@@ -0,0 +1,53 @@
+#ifndef SERENITY_SLACK_RESOURCE_HPP
+#define SERENITY_SLACK_RESOURCE_HPP
+
+#include <memory>
+#include <string>
+#include <unordered_set>
+#include <vector>
+
+#include "mesos/mesos.hpp"
+
+#include "stout/result.hpp"
+
+#include "serenity/executor_set.hpp"
+#include "serenity/serenity.hpp"
+
+namespace mesos {
+namespace serenity {
+
+/**
+ * SlackResourceObserver observes incoming ResourceUsage
+ * and produces Resource with revocable flag set (Slack Resources).
+ *
+ * Currently it only counts CPU slack
+ */
+class SlackResourceObserver : public Consumer<ResourceUsage>,
+                              public Producer<Resource> {
```

Comment:
Hm, i can't find any disadvanteges of having here Resources instead of one Resource.. it will somehow help, because in Resources you have plenty of useful helper methods.. And you will be able to make a proper sink. However i know, that we don't explicitly need that right now (we have only CPU). What's your opinion? Maybe i am wrong, but consider that (: If you place Resources here, then previous comment is obsolete (: